plugins/inputs/gtfs/gtfs.go

```diff
@@ -0,0 +1,305 @@
+  # key = "${GTFS_API_KEY}"
+  ## URL for fetching vehicle positions
+  # vehicle_positions_url = "https://host.test/VehiclePositions.pb"
+  ## URL for fetching vehicle positions
+  # trip_updates_url = "https://host.test/TripUpdates.pb"
+  ## URL for fetching vehicle positions
+  # service_alerts_url = "https://host.test/ServiceAlerts.pb"
+
+  ## Optional HTTP Basic Auth Credentials
+  # username = "username"
+  # password = "pa$$word"
+
+  ## Optional TLS Config
+  # tls_ca = "/etc/telegraf/ca.pem"
+  # tls_cert = "/etc/telegraf/cert.pem"
+  # tls_key = "/etc/telegraf/key.pem"
+  ## Use TLS but skip chain & host verification
+  # insecure_skip_verify = false
+
+  ## Amount of testTimestamp allowed to complete the HTTP request
+  # timeout = "5s"
+`
+}
+
+func (g *GTFS) Init() error {
+	tlsCfg, err := g.ClientConfig.TLSConfig()
+	if err != nil {
+		return err
+	}
+
+	g.client = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: tlsCfg,
+			Proxy:           http.ProxyFromEnvironment,
+		},
+		Timeout: g.Timeout.Duration,
+	}
+
+	if g.VehiclePositionsURL == "" && g.TripUpdatesURL == "" && g.ServiceAlertsURL == "" {
+		return errNoSourceConfigured
+	}
+
+	if g.VehiclePositionsURL != "" {
+		r, err := g.newRequest(g.VehiclePositionsURL)
+		if err != nil {
+			return err
+		}
+		g.vehiclePositionsReq = r
+	}
+
+	if g.TripUpdatesURL != "" {
+		r, err := g.newRequest(g.TripUpdatesURL)
+		if err != nil {
+			return err
+		}
+		g.tripUpdatesReq = r
+	}
+
+	if g.ServiceAlertsURL != "" {
+		r, err := g.newRequest(g.ServiceAlertsURL)
+		if err != nil {
+			return err
+		}
+		g.serviceAlertsReq = r
+	}
+
+	return nil
+}
+
+func (g *GTFS) Gather(acc telegraf.Accumulator) error {
+	var (
+		work errgroup.Group
+		now  = time.Now()
+	)
+
+	if g.vehiclePositionsReq != nil {
+		work.Go(func() error { return g.gatherVehiclePositions(acc, now) })
+	}
+
+	if g.tripUpdatesReq != nil {
+		work.Go(func() error { return g.gatherTripUpdates(acc, now) })
+	}
+
+	if g.serviceAlertsReq != nil {
+		work.Go(func() error { return g.gatherServiceAlerts(acc, now) })
+	}
+
+	return work.Wait()
+}
+
+func (g *GTFS) gatherVehiclePositions(acc telegraf.Accumulator, t time.Time) error {
+	entities, err := g.gather(g.vehiclePositionsReq)
+	if err != nil {
+		return err
+	}
+
+	for _, entity := range entities {
+		if entity.Vehicle == nil {
+			continue
+		}
+		v := entity.Vehicle
+
+		b, err := json.Marshal(v)
+		if err != nil {
+			return err
+		}
+
+		fields := map[string]interface{}{
+			"latitude":      v.Position.Latitude,
+			"longitude":     v.Position.Longitude,
+			"bearing":       v.Position.Bearing,
+			"odometer":      v.Position.Odometer,
+			"speed":         v.Position.Speed,
+			"vehicle_id":    v.Vehicle.Id,
+			"vehicle_label": v.Vehicle.Label,
+			"trip_id":       v.Trip.TripId,
+			"direction_id":  v.Trip.DirectionId,
+			"stop_id":       v.StopId,
+			"status":        v.CurrentStatus,
+			"congestion":    v.CongestionLevel,
+			"json":          b,
```

Comment:
Why do you add the whole JSON here? Can we please remove this field.

Reply:
The raw JSON contains many attributes not explicitly captured as InfluxDB fields or tags. Adding the raw JSON entity here provides flexibility in case it is necessary to inspect attributes of the entity that are not captured as first-class tags or fields.

Reply:
I disagree here. You force almost everybody to either swallow this (and live with a cluttered DB) or to drop the field requiring additional config effort. If you require this information, I prefer to print it in a debug output or provide a flag to enable this additional field.